packaging/release.py

```diff
@@ -0,0 +1,1410 @@
+
+    # TODO: consider using development versions instead of post versions after a release is published
+
+    pre = pre or ""
+    micro = version.micro
+
+    if version.is_devrelease:
+        # The next version of a development release is the same version without the development component.
+        if final:
+            pre = ""
+        elif not pre and version.pre is not None:
+            pre = f"{version.pre[0]}{version.pre[1]}"
+    elif version.is_postrelease:
+        # The next version of a post release is the next pre-release *or* micro release component.
+        if final:
+            pre = ""
+        elif not pre and version.pre is not None:
+            pre = f"{version.pre[0]}{version.pre[1] + 1}"
+
+        if version.pre is None:
+            micro = version.micro + 1
+    else:
+        raise ApplicationError(f"Version {version} is not a development or post release version.")
+
+    version = f"{version.major}.{version.minor}.{micro}{pre}"
+
+    return get_ansible_version(version, mode=mode)
+
+
+def check_ansible_version(current_version: Version, requested_version: Version) -> None:
+    """Verify the requested version is valid for the current version."""
+    if requested_version.release[:2] != current_version.release[:2]:
+        raise ApplicationError(f"Version {requested_version} does not match the major and minor portion of the current version: {current_version}")
+
+    if requested_version < current_version:
+        raise ApplicationError(f"Version {requested_version} is older than the current version: {current_version}")
+
+    # TODO: consider additional checks to avoid mistakes when incrementing the release version
+
+
+def set_ansible_version(current_version: Version, requested_version: Version) -> None:
+    """Set the current ansible-core version."""
+    check_ansible_version(current_version, requested_version)
+
+    if requested_version == current_version:
+        return
+
+    display.show(f"Updating version {current_version} to {requested_version} ...")
+
+    current = ANSIBLE_RELEASE_FILE.read_text()
+    updated = ANSIBLE_VERSION_PATTERN.sub(ANSIBLE_VERSION_FORMAT.format(version=requested_version), current)
+
+    if current == updated:
+        raise Exception("Failed to set the ansible-core version.")
+
+    ANSIBLE_RELEASE_FILE.write_text(updated)
+
+
+def test_built_artifact(path: pathlib.Path) -> None:
+    """Test the specified built artifact by installing it in a venv and running some basic commands."""
+    with tempfile.TemporaryDirectory() as temp_dir_name:
+        temp_dir = pathlib.Path(temp_dir_name)
+
+        venv_dir = temp_dir / "venv"
+        venv_bin_dir = venv_dir / "bin"
+
+        venv.create(venv_dir, with_pip=True)
+
+        env = os.environ.copy()
+        env.pop("PYTHONPATH", None)  # avoid interference from ansible being injected into the environment
+        env.update(
+            PATH=f"{venv_bin_dir}:{env['PATH']}",
```

Comment:
Nit: `os.pathsep` is a bit more generic:
```suggestion
            PATH=os.pathsep.join((venv_bin_dir, env['PATH'])),
```

Reply:
I've used `os.pathsep` in the relevant places.